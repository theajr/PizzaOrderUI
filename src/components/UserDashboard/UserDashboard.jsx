import React, { useEffect } from 'react';
import { Router, Link } from '@reach/router';
import { protectedComponent } from '../Protected';
import { connect } from '../../util/appContext';
import PizzaItemsList from '../PizzaItemsList/PizzaItemsList';
import Profile from '../Profile/Profile';
import Checkout from '../Checkout';
import MyOrders from '../MyOrders/MyOrders';

const UserDashboard = props => {
  console.log('DASH board', props);
  const { profile = {} } = props;

  return (
    <div>
      <Router>
        <PizzaItemsList path="/" />
        <Profile path="profile" />
        <Checkout path="checkout" />
        <MyOrders path="my-orders" />
      </Router>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    isLoggedIn: state?.auth?.access_token ? true : false,
    profile: state?.auth?.profile,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    addToCart: item => dispatch({ type: 'ADD_TO_CART', payload: item }),
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(protectedComponent(UserDashboard));
