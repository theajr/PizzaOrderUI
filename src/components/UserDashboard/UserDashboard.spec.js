import React from 'react';
import { shallow } from 'enzyme';
import UserDashboard from './UserDashboard';
import { renderComp } from '../../../tests/utilRender';

describe('<UserDashboard />', () => {
  test('renders', () => {
    const wrapper = renderComp(<UserDashboard />);
    expect(wrapper).toMatchSnapshot();
  });
});
