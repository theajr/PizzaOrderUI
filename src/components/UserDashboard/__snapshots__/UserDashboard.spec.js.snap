// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDashboard /> renders 1`] = `ShallowWrapper {}`;
